ejb/WFLY-6892-ejb-access-logging.adoc

```diff
@@ -0,0 +1,387 @@
+
+** long : date time ip user ejb method invocation event
+
+*** For example: 2019-05-05 12:23:27,003 127.0.0.1 user1 hello/helloBean hello 1234 received
+                 2019-05-05 12:23:30,003 127.0.0.1 user1 hello/helloBean hello 1234 finished
+
+** default: short
+
+* for more customization, define which attributes to include in ejb access logging with the following attributes:
+
+** date:
+** time
+** ip: client ip address
+** user: authenticated caller user name
+** ejb: ejb name (may need to prefix with application name)
+** method: ejb method
+** invocation: invocation id
+** event: event type ("received", "failed", "completed"; subject to available events from ejb container)
+** host: client host name
+** port: client port number
+** protocol: protocol used for this invocation
+** thread: thread name for this invocation
+** server: local server name
+
+* For example, one may configure the following custom pattern, which can be applied to both pattern formatter or
+JSON formatter:
+
+** pattern="date time client ejb method server thread"
+
+
+=== New elements in wildfly-ejb3_6_0.xsd schema
+ejb30 subsystem is extended with a new top-level element access-logging, defined as follows:
+[source,xml]
+----
+    <xs:element name="access-logging" type="accessLoggingType" minOccurs="0" maxOccurs="1" />
+
+    <xs:complexType name="accessLoggingType">
+            <xs:choice minOccurs="0" maxOccurs="unbounded">
+                <xs:element name="console-access-log" type="consoleAccessLogType"/>
+                <xs:element name="file-access-log" type="fileAccessLogType"/>
+                <xs:element name="periodic-rotating-file-access-log" type="periodicFileAccessLogType"/>
+                <xs:element name="size-rotating-file-access-log" type="sizeFileAccessLogType"/>
+                <xs:element name="server-log-access-log" type="serverLogAccessLogType"/>
+                <xs:element name="formatter" type="formatterType"/>
+            </xs:choice>
+        </xs:complexType>
+
+        <xs:complexType name="consoleAccessLogType">
+            <xs:annotation>
+                <xs:documentation>
+                    Defines an access log which writes to the console.
+                </xs:documentation>
+            </xs:annotation>
+            <xs:all>
+                <xs:element name="encoding" type="valueType" minOccurs="0"/>
+                <xs:element name="formatter" type="accessLogFormatterType" minOccurs="0"/>
+                <xs:element name="target" minOccurs="0">
+                    <xs:complexType>
+                        <xs:attribute name="name" use="required">
+                            <xs:simpleType>
+                                <xs:restriction base="xs:token">
+                                    <xs:enumeration value="System.out"/>
+                                    <xs:enumeration value="System.err"/>
+                                    <xs:enumeration value="console"/>
```

Comment:
If we use this option we will need some changes to the API provided for access logging [WFCORE-4355](https://issues.jboss.org/browse/WFCORE-4355). The API only writes to `System.out` for now.

Reply:
I will remove this option and just directly use System.out for console-access-log.

Reply:
For now we could even not use an option if we it's a separate resource.